frontend/src/services/mock_profile_backend.ts

```diff
@@ -0,0 +1,69 @@
+import { EMAILS } from "../utils/random_data";
+import { NICKNAMES } from "../utils/random_data";
+import { PRONOUNS } from "../utils/random_data"
+import { pickRandom } from "../utils/random_data";
+
+export interface PersonProps {
+  email: string;
+  nickname: string;
+  pronouns?: string; 
+  userId: string;
+}
+
+/** Error thrown when an invalid user ID is queried. */
+export class InvalidUserIdError extends Error {}
+
+/**
+ * Mimics the functionality of a Java Servlet that fetches user information 
+ * from a database based on a specified ID. 
+ */
+export class MockProfileBackendService {
+  private mockProfiles: PersonProps[] = [];
+  private readonly changeListeners: Array<(val: PersonProps[]) => void> = [];
```

Comment:
nit: This isn't used yet, so remove it. We might not ever need it in the future and we don't want dead code lying around